src/main/java/org/example/calculator/model/Calculator.java

```diff
@@ -0,0 +1,47 @@
+package org.example.calculator.model;
+
+import java.util.Arrays;
+import java.util.function.Function;
+
+public class Calculator {
+
+    private static final String DEFAULT_DELIMITER = ",|:";
+    private static final String CUSTOM_DELIMITER_REGEX = "//(.)\n";
+    private static final String CUSTOM_OPERANDS_REGEX = "//.\n(.+)";
+    private final String string;
+
+    public Calculator(String string) {
+        this.string = string == null ? "" : string;
+    }
+
+    public int calculate() {
+        String delimiter = CalculatorStringUtils.extractDelimiterWithRegex(string, CUSTOM_DELIMITER_REGEX, DEFAULT_DELIMITER);
+        String operandString = CalculatorStringUtils.extractOperandStringWithRegex(string, CUSTOM_OPERANDS_REGEX);
+        String[] operands = splitOperandStringWithDelimiter(operandString, delimiter);
+
+        return getStringOperandsSum(operands);
+    }
+
+    private int getStringOperandsSum(String[] operands) {
+        return Arrays.stream(operands)
+                .map(emptyStringToZero())
+                .map(Integer::parseInt)
+                .peek(this::validateNonNegativeInteger)
+                .mapToInt(Integer::intValue)
+                .sum();
+    }
+
+    private Function<String, String> emptyStringToZero() {
+        return s -> s.isBlank() ? "0" : s;
+    }
+
+    private String[] splitOperandStringWithDelimiter(String operandString, String delimiter) {
+        return operandString.split(delimiter);
+    }
+
+    private void validateNonNegativeInteger(Integer integer) {
+        if (integer < 0) {
+            throw new IllegalArgumentException(String.format("negatives not allowed: %d", integer));
```

Comment:
간단히 문자열 조합으로 조합할 수 있는 지금의 케이스는 그냥 문자열을 조합하는 것이 더 효율적일 것 같아요. 👀

```suggestion
            throw new IllegalArgumentException("negatives not allowed: " +  integer);
```

`String.format`은 내부적으로 `Formatter` 객체를 생성하여 format할 떄,  생각보다 이런저런 많은 연산들을 거치기 때문에 문자열 조합으로 풀기힘든 상황에 쓰는게 좋다고 생각해요. 🙂

<img width="618" alt="image" src="https://github.com/next-step/java-racingcar-kakao/assets/47850258/024dee39-472e-45dc-896c-cecd081319de">

<img width="724" alt="image" src="https://github.com/next-step/java-racingcar-kakao/assets/47850258/6a2c4272-7954-40b4-badd-51316ef2808f">